Remove ALLOW_NONLITERAL_FORMAT_BEGIN/END macros for generateHtml() in SOAuthorizationTests.mm

ddkilzer opened the pull request:
#### eacdac1d58d8a1662211a1d3de88978645e48750
<pre>
Remove ALLOW_NONLITERAL_FORMAT_BEGIN/END macros for generateHtml() in SOAuthorizationTests.mm
&lt;<a href="https://bugs.webkit.org/show_bug.cgi?id=298073">https://bugs.webkit.org/show_bug.cgi?id=298073</a>&gt;
&lt;<a href="https://rdar.apple.com/159403033">rdar://159403033</a>&gt;

Reviewed by Darin Adler.

* Source/WTF/wtf/Assertions.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
- Change static C-string declarations from `const char*` to
  `constexpr auto` so they may be used as format strings.
(generateHtml): Rename to generateHTML().
(generateHTML):
- Reduce argument list to template and one String.
- Add WTF_ATTRIBUTE_PRINTF_MATCHES() macro.
(generateOpenerHTML): Add.
- Make a copy of generateHtml() that uses a three-placeholder format
  string for use with `openerTemplate`.
- Add WTF_ATTRIBUTE_PRINTF_MATCHES() macro.
- Move `ALLOW_NONLITERAL_FORMAT_BEGIN`/`END` macros outside of both
  functions as a workaround for older Apple OSes that don&apos;t support the
  `format_matches` attribute.
(TestWebKitAPI::TEST(SOAuthorizationPopUp, NoInterceptions)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, NoInterceptionsSubFrame)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionError)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionCancel)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedCloseByItself)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedCloseByParent)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedCloseByWebKit)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedWithOtherHttpStatusCode)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedWithCookie)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedTwice)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedSuppressActiveSession)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedNewWindowNavigation)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, AuthorizationOptions)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, SOAuthorizationLoadPolicyIgnore)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, SOAuthorizationLoadPolicyAllowAsync)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, SOAuthorizationLoadPolicyIgnoreAsync)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, NoInterceptions)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, NoInterceptionsNonAppleFirstPartyMainFrame)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionError)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionCancel)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSuccess)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithOtherHttpStatusCode)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithCookie)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithCookieButCSPDeny)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithCookieButXFrameDeny)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSuccessTwice)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, AuthorizationOptions)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, SOAuthorizationLoadPolicyIgnore)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, SOAuthorizationLoadPolicyAllowAsync)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, SOAuthorizationLoadPolicyIgnoreAsync)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionErrorWithReferrer)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionErrorMessageOrder)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSuccessMessageOrder)):
- Switch from calling `generateHtml(openerTemplate, ...)` to calling
  `generateOpenerHTML(openerTemplate, ...)`.
- Rename `generateHtml()` to `generateHTML()`.

Canonical link: <a href="https://commits.webkit.org/299352@main">https://commits.webkit.org/299352@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/62926d1ff47f08cede18c4bae8b3e920db424710

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118714 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/38395 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/29046 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124897 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70774 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/08821591-86ff-478c-b4db-dca84c02e2bc) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/120592 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/39091 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46977 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/90099 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/59626 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/168aaa95-5dd9-4ae2-83d0-34cd1f8d78c3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121667 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/31142 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/106434 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70605 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bfd53e52-2cd5-4c70-b7f1-5b618416edf6) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/30206 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/24547 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/68556 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/110832 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100583 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24735 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127953 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/117228 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/45621 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/34433 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98745 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45985 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102654 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98527 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43967 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21969 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/42129 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18915 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/45491 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/51169 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/145924 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44955 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/37519 "Found 1 new JSC binary failure: testapi, Found 18604 new JSC stress test failures: ChakraCore.yaml/ChakraCore/test/Array/array_slice.js.default, ChakraCore.yaml/ChakraCore/test/Array/push2.js.default, ChakraCore.yaml/ChakraCore/test/Array/reverse1.js.default, ChakraCore.yaml/ChakraCore/test/Array/toLocaleString.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/blue_1086262.js.default, ChakraCore.yaml/ChakraCore/test/Bugs/bug215238.shrua-2.js.default, ChakraCore.yaml/ChakraCore/test/Date/date_cache_bug.js.default, ChakraCore.yaml/ChakraCore/test/Error/CallNonFunction.js.default, ChakraCore.yaml/ChakraCore/test/Error/stack2.js.default, ChakraCore.yaml/ChakraCore/test/Function/FuncBody.bug227901.js.default ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/48301 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/46641 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
```diff
@@ -68,7 +68,7 @@
 static RetainPtr<WKWebView> gNewWindow;
 
-static const char* openerTemplate =
+static const char* const openerTemplate =
```
I probably would have used `constexpr`:
```suggestion
static constexpr auto openerTemplate =
```

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
```diff
@@ -441,15 +441,31 @@ static void configureSOAuthorizationWebView(TestWKWebView *webView, TestSOAuthor
+#endif
+
+WTF_ATTRIBUTE_PRINTF_MATCHES(1, "%s")
+static String generateHtml(const char* templateHtml, const String& substitute)
```
```suggestion
static String generateHTML(const char* templateHTML, const String& substitute)
```

darinadler commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
```diff
@@ -441,15 +441,31 @@ static void configureSOAuthorizationWebView(TestWKWebView *webView, TestSOAuthor
+}
+
+WTF_ATTRIBUTE_PRINTF_MATCHES(1, "%s %s %s")
+static String generateOpenerHtml(const char* templateHtml, const String& substitute, const String& optionalSubstitute1 = emptyString(), const String& optionalSubstitute2 = emptyString())
```
```suggestion
static String generateOpenerHTML(const char* templateHTML, const String& substitute, const String& optionalSubstitute1 = emptyString(), const String& optionalSubstitute2 = emptyString())
```

Commit:
Remove ALLOW_NONLITERAL_FORMAT_BEGIN/END macros for generateHtml() in SOAuthorizationTests.mm
<https://bugs.webkit.org/show_bug.cgi?id=298073>
<rdar://159403033>

Reviewed by Darin Adler.

* Source/WTF/wtf/Assertions.h:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SOAuthorizationTests.mm:
- Change static C-string declarations from `const char*` to
  `constexpr auto` so they may be used as format strings.
(generateHtml): Rename to generateHTML().
(generateHTML):
- Reduce argument list to template and one String.
- Add WTF_ATTRIBUTE_PRINTF_MATCHES() macro.
(generateOpenerHTML): Add.
- Make a copy of generateHtml() that uses a three-placeholder format
  string for use with `openerTemplate`.
- Add WTF_ATTRIBUTE_PRINTF_MATCHES() macro.
- Move `ALLOW_NONLITERAL_FORMAT_BEGIN`/`END` macros outside of both
  functions as a workaround for older Apple OSes that don't support the
  `format_matches` attribute.
(TestWebKitAPI::TEST(SOAuthorizationPopUp, NoInterceptions)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, NoInterceptionsSubFrame)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionError)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionCancel)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedCloseByItself)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedCloseByParent)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedCloseByWebKit)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedWithOtherHttpStatusCode)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedWithCookie)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedTwice)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedSuppressActiveSession)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, InterceptionSucceedNewWindowNavigation)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, AuthorizationOptions)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, SOAuthorizationLoadPolicyIgnore)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, SOAuthorizationLoadPolicyAllowAsync)):
(TestWebKitAPI::TEST(SOAuthorizationPopUp, SOAuthorizationLoadPolicyIgnoreAsync)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, NoInterceptions)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, NoInterceptionsNonAppleFirstPartyMainFrame)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionError)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionCancel)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSuccess)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithOtherHttpStatusCode)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithCookie)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithCookieButCSPDeny)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSucceedWithCookieButXFrameDeny)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSuccessTwice)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, AuthorizationOptions)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, SOAuthorizationLoadPolicyIgnore)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, SOAuthorizationLoadPolicyAllowAsync)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, SOAuthorizationLoadPolicyIgnoreAsync)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionErrorWithReferrer)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionErrorMessageOrder)):
(TestWebKitAPI::TEST(SOAuthorizationSubFrame, InterceptionSuccessMessageOrder)):
- Switch from calling `generateHtml(openerTemplate, ...)` to calling
  `generateOpenerHTML(openerTemplate, ...)`.
- Rename `generateHtml()` to `generateHTML()`.

Canonical link: https://commits.webkit.org/299352@main
